### PR TITLE
Hotfix/validators

### PR DESCRIPTION
**Pull request by shroomist**

### CODE REVIEW

File: src/dto/proposals-response.js [hunk 1]
```diff
@@ -16,20 +16,22 @@
...
+  proposals: Array<ProposalDTO>
+}
 
-  constructor (responseData: ResponseMapped) {
-    if (typeof responseData.proposals !== 'undefined') {
-      this.proposals = responseData.proposals.map((proposal) => new ProposalDTO(proposal))
-    }
-  }
+function parseProposalsResponseDTO (responseData: ResponseMapped): ProposalsResponseDTO {
+  validate('ProposalResponseDTO', responseData, { name: 'proposals', type: 'array' })
```
Comment (donce) [inline src/dto/proposals-response.js]: !! `responseData : ResponseMapped` param indicates that this function expects to get `ResponseMapped` data, but it doesn't need to - it validates that this param has `proposals` array, so it should receive `mixed`/`any`/`Object`

File: src/validation.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+// @flow
```
Comment (donce) [inline src/validation.js]: !! Please setup IDE to add copyrights for new files:
https://mysteriumnetwork.atlassian.net/wiki/spaces/MYS/pages/238682113/Copyright+headers+setup

I understand CI should be in place to ensure that, but we'll have to setup our IDEs either way :)

File: test/unit/dto/proposal.spec.js [hunk 3]
```diff
@@ -15,41 +15,75 @@
...
-
-    it('sets wrong properties structure', async () => {
-      const proposal = new ProposalDTO('I am wrong')
-
-      expect(proposal.id).to.be.undefined
-      expect(proposal.providerId).to.be.undefined
-      expect(proposal.serviceType).to.be.undefined
-      expect(proposal.serviceDefinition).to.be.undefined
-    })
+describe('parseProposalDTO', () => {
```
Comment (donce) [inline test/unit/dto/proposal.spec.js]: !!
```suggestion
describe('.parseProposalDTO', () => {
```

File: test/unit/dto/proposals-response.spec.js [hunk 4]
```diff
@@ -15,29 +15,28 @@
...
-describe('TequilapiClient DTO', () => {
-  describe('ProposalsResponseDTO', () => {
-    it('sets properties with full structure', async () => {
-      const response = new ProposalsResponseDTO({
-        proposals: [{ id: 100 }]
-      })
-
-      expect(response.proposals).to.have.lengthOf(1)
-      expect(response.proposals[0].id).to.equal(100)
+describe('parseProposalsResponseDTO', () => {
```
Comment (donce) [inline test/unit/dto/proposals-response.spec.js]: !!
```suggestion
describe('.parseProposalsResponseDTO', () => {
```

File: test/unit/dto/proposals-response.spec.js [hunk 5]
```diff
@@ -15,29 +15,28 @@
...
-      const response = new ProposalsResponseDTO('I am wrong')
+    expect(error.message).to.eql('ProposalResponseDTO: proposals is not provided')
+  })
 
-      expect(response.proposals).to.be.undefined
-    })
+  it('throws an error if proposal in array does not validate', async () => {
+    const error = captureError(() => parseProposalsResponseDTO({ proposals: [{}] }))
+
+    expect(error.message).to.eql('ProposalDTO: id is not provided')
```
Comment (donce) [inline test/unit/dto/proposals-response.spec.js]: Nice specs 👍

File: src/dto/proposals-response.js (hunk 1 above)
Comment (ignasbernotas) [reply src/dto/proposals-response.js]: Will approve once this is fixed:)

**Comment by shroomist** [comment]
@ignasbernotas @donce please review, comments were addressed

**Pull request merged**